compiler/src/dotty/tools/io/Path.scala

```diff
@@ -127,7 +127,15 @@ class Path private[io] (val jpath: JPath) {
    * @return The path of the parent directory, or root if path is already root
    */
   def parent: Directory = jpath.normalize.getParent match {
-    case null => Directory(jpath)
+    case null =>
+      if (isAbsolute)  // it should be a root
+        toDirectory
+      else if (segments.nonEmpty && segments.last == "..")
+        (this / "..").toDirectory
+      else path match {
+        case "" | "." => Directory("..")
+        case _        => Directory(".")
+      }
     case parent => Directory(parent)
```

Comment:
Just noticed a case that is still broken because it takes this `case parent` path:

```scala
scala> Path("../..").parent
val res0: dotty.tools.io.Directory = ..      // should be ../../..
```